test/mlir/ngraph_dialect/memory_opt.mlir

```diff
@@ -0,0 +1,205 @@
+// RUN: ngraph-opt %s --split-input-file --ng-memory-opt --ngraph-memory-opt-concat --ngraph-memory-opt-eltwise | FileCheck %s
```

Comment:
Would it make sense to have concat and eltwise optimizations in separate passes. They seem two independent passes that we could be applied, customized (flags) and test independently.

Reply:
Not really. Why make separate passes over the same IR ?

Reply:
Basically they are different transformations. It's better to keep things simple and decoupled since transformations usually get complicated over time. Similar examples in MLIR: tiling, fusion, vectorization... all of them separate passes.

Reply:
Nothing in this pass is a transformation. The analysis required for both ops is the same. The only difference is that src operands for concat get a non-zero offset.